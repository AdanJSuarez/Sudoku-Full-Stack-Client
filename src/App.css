/* .App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}*/

.Image-spinning {
  height: 50px;
  opacity: 0.6;
  animation: zen-spinning infinite 500ms linear;
}
@keyframes zen-spinning {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
} 

.App {
  text-align: center;
  margin: 20px;
  font-size: 48px;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}
.Name {
  color: red
}
.Table {
  /* display: inline-block; */
  border-spacing: 0;
  border-collapse: collapse;
  margin: 0;
}
.td {
  padding:0px;
  margin: 0;
}
.Element {
  display: inline-block;
  /* position: relative; */
  padding: 5px;
  border: 2px solid grey;
}
Element:hover{
  background-color:white;
  color:black;
}