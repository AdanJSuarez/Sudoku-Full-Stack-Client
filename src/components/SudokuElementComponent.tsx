/** 
 * Adan J. Suarez
 * adanjsuarez@gmail.com
 * Full Stack Home-Project 
 */

import * as React from 'react';

export interface SudokuElementProps {
    row: number;
    column: number
    dataElement: number;
    toggleNumber: any;
}
export interface SudokuElementState {
    currentRow: number;
    currentColumn: number;
    currentDataElement: number;
    toggleOn: boolean;
}
export default class SudokuElementComponent extends React.Component<SudokuElementProps, SudokuElementState> {
    constructor(props: SudokuElementProps) {
        super(props);
        this.state = {
            currentRow: props.row,
            currentColumn: props.column,
            currentDataElement: props.dataElement,
            toggleOn: false
        };
        this.hundleClick = this.hundleClick.bind(this);
    }
    public render() {
        return (
            <button className={'Element'} onClick={this.hundleClick}> { this.state.currentDataElement } </button>
        );
    }
    /**
     * Handle click bubbling up Json object selected with click
     * @private
     * @param {React.MouseEvent<HTMLButtonElement>} e
     * @memberof SudokuElementComponent
     */
    private hundleClick(e: React.MouseEvent<HTMLButtonElement>){
        let emptyElement = {};
        let elementBubbleUp: any;
        let element = {
            number: this.state.currentDataElement,
            row: this.state.currentRow,
            column: this.state.currentColumn
        }
        this.state.toggleOn? this.setState({ toggleOn: false }) : this.setState({ toggleOn: true })
        this.state.toggleOn? elementBubbleUp = emptyElement:elementBubbleUp = element;
        this.props.toggleNumber(elementBubbleUp);
    }
}
