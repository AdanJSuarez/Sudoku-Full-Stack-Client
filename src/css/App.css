/** 
 * Adan J. Suarez
 * adanjsuarez@gmail.com
 * Full Stack Home-Project 
 */

.Image-spinning {
    height: 39px;
    opacity: 0.6;
    animation: zen-spinning infinite 500ms linear;
}
@keyframes zen-spinning {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
} 
.Name {
    font: 'arial';
    font-size: 40px;
    color: black;
}
.Reload-button{
    font-size: 25px;
}
.Table {
    /* display: inline-block; */
    border-spacing: 0;
    border-collapse: collapse;
    margin: 0;
}
.td {
    padding:0px;
    margin: 0;
}
.Element {
    height: 50px;
    width: 50px;
    font-size: 30px;
    font-style: bold;
    background-color: white;
    display: inline-block;
    padding: 5px;
    border: 2px solid grey;
}
.Element:hover{
    background-color:orange;
}
.Element:focus {
    font-size: 30px;
    border: 2px solid grey;
    background-color: grey;
    width: 50px;
    height: 50px;
}