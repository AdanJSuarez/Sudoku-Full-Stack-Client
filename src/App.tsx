/** 
 * Adan J. Suarez
 * adanjsuarez@gmail.com
 * Full Stack Home-Project 
 */

import * as React from 'react';
import './css/App.css';
import SudokuComponent from './components/Sudoku';
import zenImage from './zenForLoading.png';


interface ISudokuProps {
  sudokuNumbers?: number[];
}

interface ISudokuState {
  currentSudokuNumbers: number[];
  loading: boolean;
}

class App extends React.Component<ISudokuProps, ISudokuState> {
  
    private static url: string = 'http://127.0.0.1:8080/sudoku/board';

    private selectedNumber: any;
    
    constructor (props: ISudokuProps){
        super(props);
        this.state = {
          currentSudokuNumbers: [],
          loading: true
        }
        this.selectedNumber = 0;
        this.getSudoku();
        // Binding "this"
        this.getSudoku = this.getSudoku.bind(this);
        this.handleClick = this.handleClick.bind(this);
        this.toggledNumber = this.toggledNumber.bind(this);
        this.postNumber = this.postNumber.bind(this);
    }
    public render() {
        // Conditional render to allow zenImage shows up
        return (!this.state.loading) ? (
            <div className='App-div'>
                <h2 className ='Name'>SUDOKU</h2> 
                <button className='Reload-button' onClick={ this.handleClick }> Reload </button> 
                <SudokuComponent sudokuNumbers={this.state.currentSudokuNumbers} toggledNumber={this.toggledNumber} />
            </div>
        ) : (
            <div className='App-div'>
                <h2 className ='Name'>SUDOKU</h2> 
                <img src = {zenImage} className='Image-spinning' alt="Spinning image"/>
            </div>
        )
    }
    /**
     * Reaload a new sudoku after click
     * @private
     * @param {React.MouseEvent<HTMLButtonElement>} e
     * @memberof App
     */
    private handleClick(e: React.MouseEvent<HTMLButtonElement>) {
        this.setState({loading:true});
        if (this.selectedNumber !== 0) {
            this.postNumber();
        }
        else {
            this.getSudoku();
        }
    }
    /**
     * Take the JSON object bubbled up
     * @private
     * @param {*} element
     * @memberof App
     */
    private toggledNumber(element: any):void {
        this.selectedNumber = element;
    }
    /**
     * Fetch sudoku from the server and update state with its numbers
     * @private
     * @memberof App
     */
    private getSudoku():void {
        fetch(App.url, {
                method: 'GET',
        })
        .then((res) => res.json())
        .then((resJson) => {
            this.setState({
                currentSudokuNumbers: resJson.numbers,
                loading: false,
            });
            console.log(resJson)
        })
        .catch((err)=>{
            console.log("Error in fetch indicated down: ")
            console.error(err);
        })
    }
    /**
     *POST the number selected to the server to generate a new sudoku 
         with selected number in the same position
     * @private
     * @memberof App
     */
    private postNumber():void {
        console.log('The selected number is:',this.selectedNumber)
        fetch(App.url, {
            method: 'POST',
            body: JSON.stringify(this.selectedNumber),
        })
        .then((res) => res.json())
        .then((resJson) => {
            console.log("POST response: ", resJson)
            this.setState({
                currentSudokuNumbers:resJson.numbers,
                loading: false
            })
        })
        .catch(res => (console.log('Error POST: ', res)))
    }
}
export default App;
