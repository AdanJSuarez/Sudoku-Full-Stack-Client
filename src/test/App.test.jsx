import React from 'react';
import Enzyme, {shallow, render, mount} from 'enzyme';
import sinon from 'sinon';
import App from '../App';
import Adapter from 'enzyme-adapter-react-16';
import {createSerializer} from 'enzyme-to-json';
import SudokuElementComponent from '../components/SudokuElementComponent';
//import { expect } from 'chai';
// import SudokuComponent from '../components/Sudoku';

//Set the default serializer for jest to be from enzyme-to-json
//this produce a more easy to read serialized format
expect.addSnapshotSerializer(createSerializer({mode: "deep"}));
Enzyme.configure({adapter: new Adapter()});

// Mocks server calls ;)
jest.mock('../services/GetSudoku.ts');
jest.mock('../services/PostNumber.ts');

describe("Test suit for App", ()=>{
    const sudokuSolved = [2,5,8,7,4,3,9,6,1,6,7,1,5,9,8,2,3,4,3,4,9,2,1,6,8,7,5,5,6,
        3,9,7,1,4,8,2,1,8,2,3,5,4,7,9,6,7,9,4,8,6,2,1,5,3,4,1,7,6,8,5,3,2,9,9,3,6,
        1,2,7,5,4,8,8,2,5,4,3,9,6,1,7];
    
    it('render correctly shallow without numbers', ()=>{
        const wrapper = shallow(
            <App sudokuNumbers={[]} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render correctly shallow with numbers', ()=>{
        const wrapper = shallow(
            <App sudokuNumbers={sudokuSolved} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render correctly render without numbers', ()=>{
        const wrapper = render(
            <App sudokuNumbers={[]} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render correctly render with numbers', () => {
        const wrapper = render(
            <App sudokuNumbers={sudokuSolved} />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('render correctly mount without numbers', () => {
        const wrapper = mount(<App sudokuNumbers={[]}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('render correctly mount with numbers', () => {
        const wrapper = mount(<App sudokuNumbers={sudokuSolved}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('Test App has a image when it has no sudoku numbers', () => {
        const wrapper = shallow(<App sudokuNumbers={[]} />);
        expect(wrapper.find('img').hasClass('Image-spinning')).toBe(true);
    });
    it('Test load number after first render', () => {
        const wrapper = shallow(<App sudokuNumbers={[]} />);
        expect(wrapper.find('img').hasClass('Image-spinning')).toBe(true);
        wrapper.update();
        expect(wrapper.find('img').hasClass('Image-spinning')).toBe(false);
    })
    it('Test App has no image when it has no sudoku numbers and has a button', ()=>{
        const wrapper = shallow(<App sudokuNumbers={sudokuSolved}/>);
        expect(wrapper.hasClass('Reload-button')).toBe(false);
        expect(wrapper.find('button').first().hasClass('Reload-button')).toBe(true); 
    });
    it('Test click on a reload with sudoku numbers', () => {
        const wrapper = shallow(<App sudokuNumbers={sudokuSolved} />); //
        expect(wrapper.find('button').hasClass('Reload-button')).toBe(true);
        expect(wrapper.state('loading')).toBe(false);
        wrapper.find('button').simulate('click');
        expect(wrapper.state('loading')).toBe(true);
    });
})