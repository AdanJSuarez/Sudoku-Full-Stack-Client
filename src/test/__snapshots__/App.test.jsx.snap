// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit for App render correctly render 1`] = `
<div
  class="App-div"
>
  <h2
    class="Name"
  >
    SUDOKU
  </h2>
  <img
    alt="Spinning image"
    class="Image-spinning"
  >
    
  </img>
</div>
`;

exports[`Test suit for App render correctly shallow 1`] = `
<div
  className="App-div"
>
  <h2
    className="Name"
  >
    SUDOKU
  </h2>
  <img
    alt="Spinning image"
    className="Image-spinning"
  />
</div>
`;
